src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java

```diff
@@ -4,71 +4,50 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStreamWriter;
-import java.io.UnsupportedEncodingException;
 import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLEncoder;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Map.Entry;
+
+import com.atlassian.confluence.json.json.JsonObject;
 
 public class FlowdockConnection {
-	public static void sendApiMessage(Map<String, String> params, String apiKey) {
+	public static void sendApiMessage(JsonObject params, String apiKey) {
 		if (params == null || apiKey == null) {
 			return;
 		}
-		
+
 		try {
-			URL apiUrl = getApiUrl(apiKey);
+			URL apiUrl = getApiUrl(URLEncoder.encode(apiKey, "UTF-8"));
 			postData(apiUrl, params);
 		} catch (MalformedURLException mue) {
-			
+
 		} catch (IOException ioe) {
-			
+
 		}
 	}
-	
-	private static void postData(URL apiUrl, Map<String, String> params) throws IOException {
-		HttpURLConnection conn = (HttpURLConnection)apiUrl.openConnection();
+
+	private static void postData(URL apiUrl, JsonObject params) throws IOException {
+		HttpURLConnection conn = (HttpURLConnection) apiUrl.openConnection();
 		conn.setRequestMethod("POST");
 		conn.setUseCaches(false);
 		conn.setDoOutput(true);
 		conn.setDoInput(true);
+		conn.setRequestProperty("Content-Type", "application/json");
 		conn.setRequestProperty("User-Agent", "Flowdock Confluence Plugin");
-		
+
 		OutputStreamWriter out = new OutputStreamWriter(conn.getOutputStream());
-		String postData = mapToPostData(params);
-		out.write(postData);
+		out.write(params.serialize());
 		out.close();
-		
+
 		BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
 		while (in.readLine() != null);
 		in.close();
-		
+
 		conn.connect();
```

Comment:
I wonder if this is really needed. It seems like we have already written/read to/from the connection. The [docs](http://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#connect()) seem to indicate that conn.connect() would be ignored if it was already opened. We might want to move this line up to right after conn.setRequestProperty.